### PR TITLE
1479353: Add --after option to list command

### PR DESCRIPTION
**Pull request by cnsnyder**
After looking at the following PR[1], it occurred to me that we did not have an implementation on the subman side for this feature.

This PR implements RFE 1479353 using the existing candlepin query parameters.

Should the candlepin PR be accepted, I will update this PR to match the new implementation.

[1]: https://github.com/candlepin/candlepin/pull/1862

### CODE REVIEW

**Comment by cnsnyder** [comment]
[1479353: Add support for looking up pools available 'after' a specified date.](https://trello.com/c/4ezql5BU/695-1479353-add-support-for-looking-up-pools-available-after-a-specified-date)

**Comment by cnsnyder** [comment]
[1479353: Add support for looking up pools available 'after' a specified date.](https://trello.com/c/4ezql5BU/695-1479353-add-support-for-looking-up-pools-available-after-a-specified-date)

**Comment by cnsnyder** [comment]
[1479353: Add support for looking up pools available 'after' a specified date.](https://trello.com/c/4ezql5BU/695-1479353-add-support-for-looking-up-pools-available-after-a-specified-date)

**Comment by cnsnyder** [comment]
[1479353: Add support for looking up pools available 'after' a specified date.](https://trello.com/c/4ezql5BU/695-1479353-add-support-for-looking-up-pools-available-after-a-specified-date)

**Comment by cnsnyder** [comment]
[1479353: Add support for looking up pools available 'after' a specified date.](https://trello.com/c/4ezql5BU/695-1479353-add-support-for-looking-up-pools-available-after-a-specified-date)

**Comment by cnsnyder** [comment]
[1479353: Add support for looking up pools available 'after' a specified date.](https://trello.com/c/4ezql5BU/695-1479353-add-support-for-looking-up-pools-available-after-a-specified-date)

**Comment by cnsnyder** [comment]
[1479353: Add support for looking up pools available 'after' a specified date.](https://trello.com/c/4ezql5BU/695-1479353-add-support-for-looking-up-pools-available-after-a-specified-date)

**Comment by cnsnyder** [comment]
[1479353: Add support for looking up pools available 'after' a specified date.](https://trello.com/c/4ezql5BU/695-1479353-add-support-for-looking-up-pools-available-after-a-specified-date)

**Comment by cnsnyder** [comment]
[1479353: Add support for looking up pools available 'after' a specified date.](https://trello.com/c/4ezql5BU/695-1479353-add-support-for-looking-up-pools-available-after-a-specified-date)

**Comment by cnsnyder** [comment]
[1479353: Add support for looking up pools available 'after' a specified date.](https://trello.com/c/4ezql5BU/695-1479353-add-support-for-looking-up-pools-available-after-a-specified-date)

**Comment by kahowell** [comment]
@cnsnyder FYI, there are some test failures. Also redundant comments with trello links... bad automation?

**Comment by cnsnyder** [comment]
Reworked the PR to just make use of the new after query param.

Cheers

File: src/rhsm/connection.py [hunk 1]
```diff
@@ -1281,7 +1281,7 @@ def checkin(self, consumerId, checkin_date=None):
 
         return self.conn.request_put(method)
 
-    def getPoolsList(self, consumer=None, listAll=False, active_on=None, owner=None, filter_string=None, future=None):
+    def getPoolsList(self, consumer=None, listAll=False, active_on=None, owner=None, filter_string=None, after=None):
         """
```
Comment (wottop) [inline src/rhsm/connection.py]: Did you replace future with after? They should both be there.

File: src/rhsm/connection.py [hunk 2]
```diff
@@ -1304,9 +1304,10 @@ def getPoolsList(self, consumer=None, listAll=False, active_on=None, owner=None,
 
         if listAll:
             method = "%s&listall=true" % method
-        if future in ('add', 'only'):
```
Comment (wottop) [inline src/rhsm/connection.py]: Same comment on removing future

Comment (wottop) [reply src/rhsm/connection.py]: If we didn't really use future properly before in SubMan, let me know.

File: src/subscription_manager/managercli.py [hunk 3]
```diff
@@ -2265,6 +2267,31 @@ def _validate_options(self):
             system_exit(os.EX_USAGE, _("Error: --no-overlap is only applicable with --available"))
         if self.options.pid_only and self.options.installed:
             system_exit(os.EX_USAGE, _("Error: --pool-only is only applicable with --available and/or --consumed"))
+        if self.options.after and not self.options.available:
+            system_exit(os.EX_USAGE, _("Error: --after is only applicable with --available"))
+        if self.options.after and not self.options.all:
+            system_exit(os.EX_USAGE, _("Error: --after is only applicable with --all"))
```
Comment (wottop) [inline src/subscription_manager/managercli.py]: this pair of conditions does not make sense. if it passes the first one, then if fails the second and vice versa

shouldn't it be compunded to one condition: 'if self.options.after and not (self.options.available or  self.options.all)'

**Comment by kahowell** [comment]
@cnsnyder If the candlepin server does not support after, what happens? We should make sure the failure is graceful and lets the user know `--after` is not supported by the server.

**Comment by wottop** [comment]
retest this please

**Comment by wottop** [comment]
retest this please

**Comment by wottop** [comment]
retest this please

**Comment by wottop** [comment]
stylish issue:

test/test_managercli.py:424:49: F841 local variable 'patcher' is assigned to but never used

**Comment by wottop** [comment]
retest this please

**Comment by wottop** [comment]
ack

**Pull request merged**